packages/oauth-adapters/src/oauthAuthenticationAdapter.ts

```diff
@@ -1,19 +1,24 @@
-import { OAuthToken } from './oAuthToken';
 import {
   AuthenticatorInterface,
   passThroughInterceptor,
 } from '@apimatic/core-interfaces';
 import { AUTHORIZATION_HEADER, setHeader } from '@apimatic/http-headers';
 import { OAuthConfiguration } from './oAuthConfiguration';
 
-export const requestAuthenticationProvider = (
-  initialOAuthToken?: OAuthToken,
-  oAuthTokenProvider?: (token: OAuthToken | undefined) => Promise<OAuthToken>,
-  oAuthOnTokenUpdate?: (token: OAuthToken) => void,
-  oAuthConfiguration?: OAuthConfiguration
+export interface OAuthTokenConstraints {
+  accessToken?: string;
+  expiry?: bigint;
+}
+
+export const requestAuthenticationProvider = <T extends OAuthTokenConstraints>(
```

Comment:
@Ayeshas09 Please refactor this function so that it only contains a maximum of 25 lines